.github/scripts/release.sh

```diff
@@ -1,57 +1,312 @@
+        --head "${SETUP_REMOTE_SLUG}:${SETUP_REMOTE_BRANCH}" \
+        --repo "BetonQuest/BetonQuest" \
+        2>&1 > /dev/null | sed 's/^/        /'
+        return
+    else
+      echo '    ! Looks like no branch is checked out. Create the pull request manually!'
+    fi
+  else
+    echo '    ! You do not have '"'GitHub CLI'"' installed. Create the pull request manually!'
+  fi
+  echo "    The changes are already in your repository ${SETUP_REMOTE_REPOSITORY} in branch ${SETUP_REMOTE_BRANCH}"
+}
+
+setupPublishCreatePullRequestSlug() {
+  SETUP_REMOTE_SLUG_RAW="$(git config --get remote.${SETUP_REMOTE_REPOSITORY}.url)"
+  if [ "${SETUP_REMOTE_SLUG_RAW:0:15}" == 'git@github.com:' ]; then
+    SETUP_REMOTE_SLUG=${SETUP_REMOTE_SLUG_RAW:15:${#SETUP_REMOTE_SLUG_RAW}-30}
+  fi
+  if [ "${SETUP_REMOTE_SLUG_RAW:0:19}" == 'https://github.com/' ]; then
+    SETUP_REMOTE_SLUG=${SETUP_REMOTE_SLUG_RAW:19:${#SETUP_REMOTE_SLUG_RAW}-30}
+  fi
+  if [ -z "${SETUP_REMOTE_SLUG}" ]; then
+    printNewSection
+    echo 'No GitHub remote slug could be extracted from remotes!'
+    exit 1
+  fi
+}
+
+#
+# Start of script
+#
+
+printHelp
+goToRootDirectory
+checkRequirements
+selectAction
+
+version
+if [ "${ACTION}" == 1 ] || [ "${ACTION}" == 2 ]; then
+  releasePrepare
+fi
+if [ "${ACTION}" == 1 ] || [ "${ACTION}" == 3 ]; then
+  setupPrepare
 fi
 
-echo ""
-echo "Enter the name (not URL) of the remote repository you want to create the release in."
-echo "'upstream' is the fallback if no input is provided!"
-echo -n "Remote repository: "
-read -r REMOTE_REPOSITORY
-if [ -z "${REMOTE_REPOSITORY}" ]; then
-  REMOTE_REPOSITORY=upstream
-  echo -en "\033[1A\033[2K"
-  echo "Remote repository: upstream"
+printNewSection
+echo "Create ${ACTION_TYPE}..."
+if [ "${ACTION}" == 1 ] || [ "${ACTION}" == 2 ]; then
+  releasePublish
+fi
+if [ "${ACTION}" == 1 ] || [ "${ACTION}" == 3 ]; then
+  setupPublish
 fi
 
-echo ""
-echo "Create version tag and push it..."
-git tag v"${MAVEN_VERSION}" HEAD
-git push "${REMOTE_REPOSITORY}" v"${MAVEN_VERSION}"
-echo "Version tag created and pushed."
-
-echo ""
-echo "Update version and changelog..."
-sed -i "s~<version>${MAVEN_VERSION}</version>~<version>${NEXT_VERSION}</version>~g" pom.xml
-DATE=$(date +%Y-%m-%d)
-NEW_CHANGELOG="## \[Unreleased\] - \${current-date}\n### Added\n### Changed\n### Deprecated\n### Removed\n### Fixed\n### Security\n"
-sed -i "s~## \[Unreleased\] - \${current-date}~${NEW_CHANGELOG}\n## \[${MAVEN_VERSION}\] - ${DATE}~g" CHANGELOG.md
-echo "Version and changelog updated."
-
-echo ""
-echo "Commit and push 'pom.xml' and 'CHANGELOG.md'..."
-git commit --all --message="Version bump to ${NEXT_VERSION}"
-git push ${REMOTE_REPOSITORY}
-echo "'pom.xml' and 'CHANGELOG.md' committed and pushed."
-
-echo ""
-echo ""
-echo "Release script run was successful and a release is now in progress on GitHub."
+printNewSection
+echo "Script run was successful and a ${ACTION_TYPE} is now in progress on GitHub"
+printNewSection
```

Comment:
I wouldn't print empty lines at the end of the script, but this is more of a style question and not a strict requirement.

Reply:
Otter and i prefer empty lined and the end